src/plugins/uploader.js

```diff
@@ -92,17 +93,28 @@ class Uploader {
 		}
 
 		// Force a download in the browser if it's not an allowed type (binary or otherwise unknown)
-		const contentDisposition = inlineContentDispositionTypes.includes(detectedMimeType)
-			? "inline"
-			: "attachment";
+		let slug = req.params.slug;
+		let disposition =
+			detectedMimeType in inlineContentDispositionTypes ? "inline" : "attachment";
+
+		if (!slug && detectedMimeType in inlineContentDispositionTypes) {
+			slug = inlineContentDispositionTypes[detectedMimeType];
```

Comment:
having the same lookup twice (`detectedMimeType in inlineContentDispositionTypes`) in such a short order seems strange to me... why not extracting this to a const so that the thing is obvious?

Reply:
I don't really care tbh. Gonna change it :)